bigtable/client/internal/endian.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_INTERNAL_ENDIAN_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_INTERNAL_ENDIAN_H_
+
+#include "bigtable/client/internal/encoder.h"
+#include "bigtable/client/internal/strong_type.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+using bigendian64_t = internal::StrongType<std::int64_t, struct BigEndianType>;
+
+namespace internal {
+
```

Comment:
Please add the specialization here:

```C++
template<>
struct Decoder<bigendian64_t> {
... ...
...
};
```

I know it is working right now, but *why* it is working is not portable (it depends on how the compilers we use implement templates via weak symbols).